specification/confidentialledger/resource-manager/Microsoft.ConfidentialLedger/preview/2023-06-28-preview/confidentialledger.json

```diff
@@ -0,0 +1,849 @@
+            "name": "confidentialLedger",
+            "in": "body",
+            "description": "Confidential Ledger Backup Request Body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ConfidentialLedgerBackup"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The request was successful; response contains a message stating the Confidential Ledger Resource is successfully being backed up.",
+            "schema": {
+              "$ref": "#/definitions/ConfidentialLedgerBackupResponse"
+            }
+          },
+          "202": {
+            "description": "The Confidential Ledger Resource will be backed up asynchronously."
+          },
+          "default": {
+            "description": "Error response describing the reason for operation failure. 404 - NotFound (Confidential Ledger does not exist.)",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "ConfidentialLedgerBackup": {
+            "$ref": "./examples/ConfidentialLedger_Backup.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ConfidentialLedger/ledgers/{ledgerName}/restore": {
+      "post": {
+        "tags": [
+          "Confidential Ledger"
+        ],
+        "summary": "Performs the restore operation to spin up a newly restored Confidential Ledger Resource.",
+        "description": "Restores a Confidential Ledger Resource.",
+        "operationId": "Ledger_Restore",
+        "x-ms-long-running-operation": true,
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/LedgerNameParameter"
+          },
+          {
+            "name": "confidentialLedger",
+            "in": "body",
+            "description": "Confidential Ledger Restore Request Body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ConfidentialLedgerRestore"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The request was successful; response contains a message stating the Confidential Ledger Resource is successfully being restored.",
+            "schema": {
+              "$ref": "#/definitions/ConfidentialLedgerRestoreResponse"
+            }
+          },
+          "201": {
```

Comment:
>201 [](http://example.com/codeflow?start=11&length=3)

Why does POST return 201 instead 202 here?

Reply:
During a restore we are technically creating (re-creating) the resource once again using the backup so we should expect a 201 created response rather than a 202 updated/patch response.

Reply:
However, this is an action on an existing resource, so it can't simply create a new resource. Only PUT operations should be used to create new resources. ARM caches the resources created/updated by PUT/PATCH, so if you create a new resource with POSY, ARM cache will have out-dated copy and will not be able sync this resource.

Reply:
Understood, I have made it a 202 to follow the ARM flow.